masonry/src/tree_arena.rs

```diff
@@ -372,6 +372,38 @@ impl<'a, Item> ArenaMutChildren<'a, Item> {
     }
 }
 
+impl<'a, Item> ArenaRef<'a, Item> {
+    pub fn id(&self) -> u64 {
+        self.children.id.unwrap()
+    }
+}
+
+impl<'a, Item> ArenaMut<'a, Item> {
+    pub fn id(&self) -> u64 {
+        self.children.id.unwrap()
+    }
+
+    /// Returns a shared token equivalent to this one.
```

Comment:
`Shared` doesn't seem quite right, given that `ArenaRef` cannot be copied (etc.). That is, `ArenaRef` does not appear to have any additional abilities over `ArenaMut`, except for being passed to methods which accept `ArenaRef`.

This method also doesn't seem to be used?

Reply:
> Shared doesn't seem quite right, given that ArenaRef cannot be copied

Well now it can =P

> This method also doesn't seem to be used?

Yeah, but I'm writing this as though it was a public crate.